ticdc: update pulsar sink (#3753)
pingcap/docs: ticdc/manage-ticdc.md

| `ca` | The path of the CA certificate file needed to connect to the downstream Kafka instance (optional)  |
| `cert` | The path of the certificate file needed to connect to the downstream Kafka instance (optional) |
| `key` | The path of the certificate key file needed to connect to the downstream Kafka instance (optional) |

#### Configure sink URI with `pulsar`

Sample configuration:

{{< copyable "shell-regular" >}}

```shell
--sink-uri="pulsar://127.0.0.1:6650/cdc-test?connectionTimeout=2s"
```

The following are descriptions of parameters that can be configured for the sink URI with `pulsar`:

| Parameter  | Description                                            |
| :------------------ | :------------------------------------------------------------ |
| `connectionTimeout` | The timeout for establishing a connection to the downstream Pulsar, which is optional and defaults to 30 (seconds) |
| `operationTimeout` | The timeout for performing an operation on the downstream Pulsar, which is optional and defaults to 30 (seconds) |
| `tlsTrustCertsFilePath` | The path of the CA certificate file needed to connect to the downstream Pulsar instance (optional) |

> | `tlsAllowInsecureConnection` | Determines whether to accept untrusted TLS certificate from broker (optional) |

Reviewer: ```suggestion
| `tlsAllowInsecureConnection` | Determines whether to allow unencypted connection after TLS is enabled (optional) |
```

| `tlsValidateHostname` |  Determines whether to verify the host name of the certificate from the downstream Pulsar (optional) |
| `maxConnectionsPerBroker` | The maximum number of connections allowed to a single downstream Pulsar broker, which is optional and defaults to 1 |
| `auth.tls` | Uses the TLS mode to verify the downstream Pulsar (optional). For example, `"{"tlsCertFile":"/path/to/cert", "tlsKeyFile":"/path/to/key"}"`. |
| `auth.token` | Uses the token mode to verify the downstream Pulsar (optional). For example, `"{"token":"secret-token"}"` or `"{"file":"path/to/secret-token-file"}"`. |
| `name` | The name of Pulsar producer in TiCDC (optional) |

> | `maxPendingMessages` | Sets the maximum queue size of pending messages, which is optional and defaults to 1000. For example, pending for the confirmation message from Pulsar. |

Reviewer: ```suggestion
| `maxPendingMessages` | Sets the maximum size of the pending message queue, which is optional and defaults to 1000. For example, pending for the confirmation message from Pulsar. |
```

| `disableBatching` |  Disables automatically sending messages in batches (optional) |
| `batchingMaxPublishDelay` | Sets the duration within which the messages sent are batched (default: 10ms) |
| `compressionType` | Sets the compression algorithm used for sending messages (optional). The value options are `LZ4`, `ZLIB`, and `ZSTD` (default). |
| `hashingScheme` | The hash algorithm used for choosing the partition to which a message is sent (optional). The value options are `JavaStringHash` (default) and `Murmur3`. |

> | `properties.*` |  The customized properties added to the Pulsar producer in TiCDC (optional). For example, `properties.location=Hangzhou`. |

Reviewer: ```suggestion
| `properties.*` | The customized properties added to the Pulsar producer in TiCDC (optional). For example, `properties.location=Hangzhou`. |
```
